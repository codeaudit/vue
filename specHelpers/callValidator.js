import ReactPropTypesArgs from './reactPropTypeArgs';

const callValidator = (validator, element, propName = '', componentName = '') => validator(element.props, propName, componentName, ...ReactPropTypesArgs);

export const validatorPasses = (validator, element, propName) => callValidator(validator, element, propName) === null;
export const validatorFails = (validator, element, propName) => callValidator(validator, element, propName) instanceof Error;
