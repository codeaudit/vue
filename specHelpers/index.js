import React from 'react';
import ThemeProvider, {getTheme} from './../src/Theme';

export const applyTheme = (ComponentUnderTest, theme = getTheme()) => <ThemeProvider theme={theme}>{ComponentUnderTest}</ThemeProvider>;
