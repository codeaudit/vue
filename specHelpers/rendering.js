import toJson from 'enzyme-to-json';
import {mount, shallow} from 'enzyme';
import React from 'react';
import theme from './TestTheme';

export const getRender = (Component) => (props = {}, context = {theme}) => shallow(<Component {...props} />, {context});
export const getMount = (Component) => (props = {}, context = {theme}) => mount(<Component {...props} />, {context});
export const snapshot = (renderedComponent) => toJson(renderedComponent);
