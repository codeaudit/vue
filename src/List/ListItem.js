import EventListener from 'react-event-listener';
import React, {Component, PropTypes} from 'react';
import Radium from './../utilities/Radium';
import transparent from './../utilities/Transparent';
import {createEventHandler} from './../utilities/component';

const getStyles = (props, theme) => {
    const {
        highlighted,
    } = props;
    const {
        smallGutter,
        largeGutter,
    } = theme;
    const hoveredStyles = highlighted
        ? {
            backgroundColor: props.highlightBackgroundColor,
            color: props.highlightColor,
        } : {
            backgroundColor: transparent,
            color: 'initial',
        };

    return {
        listItem: {
            cursor: 'pointer',
            padding: `${smallGutter}px ${largeGutter}px`,
            ...hoveredStyles,
        },
    };
};

const ListItem = (props, context) => {
    const {
        children,
        itemOid,
        onKeyUp,
        onMouseEnter,
    } = props;
    const handleMouseEnter = createEventHandler(onMouseEnter, itemOid);
    const handleKeyUp = createEventHandler(onKeyUp, itemOid);
    const styles = getStyles(props, context.theme);

    return (
        <div
            style={styles.listItem}
            onMouseEnter={handleMouseEnter}
            onKeyUp={handleKeyUp}
        >
            <EventListener
                target="window"
                onKeyUp={handleKeyUp}
            />
            {children}
        </div>
    );
};
ListItem.propTypes = {
    /**
     * Content to render within the list item
     */
    children: PropTypes.node.isRequired,
    /**
     * Color of the background when in a highlighted state
     */
    highlightBackgroundColor: PropTypes.string,
    /**
     * Color of the text when in a highlighted state
     */
    highlightColor: PropTypes.string,
    /**
     * When true, indicates the component is in a highlighted state
     */
    highlighted: PropTypes.bool,
    /**
     * Event handler; fired once the item is highlighted AND a key is pressed
     */
    onKeyUp: PropTypes.func,
    /**
     * Event handler; fired once mouse enters the component
     */
    onMouseEnter: PropTypes.func,
};
ListItem.defaultProps = {
    highlighted: false,
    onClick: () => {
    },
    onMouseEnter: () => {
    },
    onKeyUp: () => {
    },
};
ListItem.contextTypes = {
    theme: PropTypes.object.isRequired,
};
ListItem.displayName = 'ListItem';

export default Radium(ListItem);
