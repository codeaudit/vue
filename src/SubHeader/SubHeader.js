import React, {PropTypes} from 'react';
import Radium from './../utilities/Radium';
import ThemeProvider from './../Theme';

const SubHeader = (props)=> {
    const {children} = props;

    return (
        <header>
            {children}
        </header>
    );
};
SubHeader.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node,
    ]),
};
SubHeader.contextTypes = {theme: PropTypes.shape(ThemeProvider.themeDefinition).isRequired,};

export default Radium(SubHeader);
