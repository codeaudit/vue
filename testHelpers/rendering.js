import toJson from 'enzyme-to-json';
import React, {PropTypes} from 'react';
import {mount, render, shallow} from 'enzyme';
import theme from './../packages/themes/src/V1Theme';

const ctx = {
    theme,
};

export const getShallow = (Component) => (props = {}, context = {}) => shallow(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
});

export const getRender = (Component) => (props = {}, context = {}) => render(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
});

export const getMount = (Component) => (props = {}, options = {}, context = {}) => mount(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
    ...options,
});

const replaceUIKey = (componentName) => {
    let index = 0;
    return (json) => {
        if (!json.props || !json.props.uiKey) {
            return json;
        }
        const uiKey = `${componentName}-${index}`;
        index += 1;
        return Object.assign({}, json, {
            uiKey,
            uiPath: [uiKey],
        });
    };
};
const replaceUiKeyRecursively = (json) => {
    if (!json.type) {
        return json;
    }
    const replaceUIKeyInParent = replaceUIKey(json.type);
    const output = replaceUIKeyInParent(json);
    if (!output.children) {
        return json;
    }
    output.children = json.children.map((child, index) => replaceUiKeyRecursively(child));
    return output;
};

export const snapshot = (renderedComponent) => {
    return toJson(renderedComponent);
};

export const reset = (component) => () => {
    if (component) {
        component.unmount();
    }
    document.body.innerHTML = '';
};
