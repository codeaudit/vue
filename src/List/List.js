import React, {Component, PropTypes} from 'react';
import ListItem from './ListItem';
import Radium from './../utilities/Radium';
import SubHeader from './../SubHeader';
import transparent from './../utilities/Transparent';
import * as CustomPropTypes from './../utilities/CustomPropTypes';

class List extends Component {
    static propTypes = {
        /**
         * ListItem or SubHeader components.
         */
        children: CustomPropTypes.oneOfComponentType([
            ListItem,
            SubHeader
        ]),
        /**
         * Background color used on hovered list items.
         */
        hoverBackgroundColor: PropTypes.string,
        /**
         * Font color used on hovered list items.
         */
        hoverColor: PropTypes.string,
    };
    static defaultProps = {
        hoverBackgroundColor: transparent,
    };

    constructor(props, ...rest) {
        super(props, ...rest);
        this.getStyles = this.getStyles.bind(this);
    }

    getStyles() {
        return {
            list: {
                backgroundColor: 'white',
                padding: `8px 0`,
            }
        };
    }

    render() {
        const {
            children,
            hoverBackgroundColor,
            hoverColor,
        } = this.props;
        const styles = this.getStyles();

        return (
            <div
                style={styles.list}>
                {React.Children
                    .map(children, (child) => {
                        if (!Boolean(child)) {
                            return;
                        }
                        return React.cloneElement(child, {
                            hoverBackgroundColor,
                            hoverColor,
                        });
                    })
                }
            </div>
        )
    }
}
export default Radium(List)
