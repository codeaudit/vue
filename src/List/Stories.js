import React from 'react';
import {action, storiesOf} from '@kadira/storybook';
import List, {ListItem} from './';
import SubHeader from './../SubHeader';

const listItems = [
    'Item ',
    'Item ',
    'Item ',
    'Item ',
]
storiesOf('List')
    .addWithInfo('basic list',
        ``,
        () => (
            <List
                highlightColor="pink"
                highlightBackgroundColor="blue"
                onSelectItem={action('selected item')}
            >
                <SubHeader>Sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{(index + 1)}
                    </ListItem>
                ))}
                <SubHeader>Another sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{index + 5}
                    </ListItem>
                ))}
            </List>
        )
    ).addWithInfo('with event listeners',
        ``,
        () => (
            <List
                highlightColor="pink"
                highlightBackgroundColor="blue"
                onItemHighlighted={action('highlighted')}
                onMouseEnter={action('mouse entered list')}
                onMouseLeave={action('mouse left list')}
            >
                <SubHeader>Sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{(index + 1)}
                    </ListItem>
                ))}
                <SubHeader>Another sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{index + 5}
                    </ListItem>
                ))}
            </List>
        )
    )
    .addWithInfo('empty list',
        ``,
        () => (
            <List />
        )
    )
    .addWithInfo('scrolling list',
        ``,
        () => (
            <List
                maxHeight={225}
                onItemHighlighted={action('highlighted')}
            >
                <SubHeader>Sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{(index + 1)}
                    </ListItem>
                ))}
                <SubHeader>Another sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{index + 5}
                    </ListItem>
                ))}
            </List>
        )
    )
    .addWithInfo('set highlighted item',
        ``,
        () => (
            <List
                highlightColor="pink"
                highlightBackgroundColor="blue"
                highlightedIndex={2}
                maxHeight={225}
                onItemHighlighted={action('highlighted')}
            >
                <SubHeader>Sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{(index + 1)}
                    </ListItem>
                ))}
                <SubHeader>Another sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{index + 5}
                    </ListItem>
                ))}
            </List>
        )
    )
    .addWithInfo('keyboard navigation enabled',
        ``,
        () => (
            <List
                active={true}
                highlightColor="pink"
                highlightBackgroundColor="blue"
                highlightedIndex={2}
                maxHeight={225}
                onItemHighlighted={action('highlighted')}
                onSelectItem={action('selected item')}
            >
                <SubHeader>Sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{(index + 1)}
                    </ListItem>
                ))}
                <SubHeader>Another sub header text</SubHeader>
                {listItems.map((item, index) => (
                    <ListItem
                        key={index}
                    >
                        {item}{index + 5}
                    </ListItem>
                ))}
            </List>
        )
    );
