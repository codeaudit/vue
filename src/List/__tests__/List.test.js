import React from 'react';
import List from './../List';
import ListItem from './../ListItem';
import SubHeader from './../../SubHeader';
import {getMount, getShallow, snapshot} from './../../../specHelpers/rendering';

const renderList = getShallow(List);
const mountList = getMount(List);
const evt = {test: true};

let component;
afterEach(() => {
    component.unmount();
    document.body.innerHTML = '';
});

test('List can render ListItems', () => {
    component = renderList({
        children: getListItems(),
    });
    expect(snapshot(component)).toMatchSnapshot();
});

test('List can have a specified highlight color for its ListItems', () => {
    component = renderList({
        children: getListItems(),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
    });
    expect(snapshot(component)).toMatchSnapshot();
});

test('List\'s highlighted item can be set', () => {
    component = renderList({
        children: getListItems(),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
        ui: {
            highlightedIndex: 1,
        },
    });
    expect(snapshot(component)).toMatchSnapshot();
});

test('List\'s list items are highlighted when hovered over', () => {
    component = renderList({
        children: getListItems(),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
    });
    simulateHover(component);
    expect(snapshot(component)).toMatchSnapshot();
});

test('List can render ListItems with scroll bar', () => {
    component = mountList({
        children: getListItems(),
        maxHeight: 50,
    });
    expect(snapshot(component)).toMatchSnapshot();
});

test('List can have a callback for when an item is highlighted', () => {
    const onHighlightItem = jest.fn();
    component = mountList({
        children: getListItems(),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
        onHighlightItem,
    });
    simulateHover(component, evt);
    expect(onHighlightItem).toHaveBeenCalledTimes(1);
    expect(onHighlightItem.mock.calls[0][0].test).toBeTruthy();
    expect(onHighlightItem.mock.calls[0][1]).toEqual(2);
});

test('Up arrow key moves highlighted item up to next highlight-able item when list is active', () => {
    const onHighlightItem = jest.fn();
    component = mountList({
        active: true,
        children: getListItems().concat(getListItems()),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
        highlightedIndex: 4,
        onHighlightItem,
    });
    simulateUpArrowKey(component);
    expect(onHighlightItem).toHaveBeenCalledTimes(1);
    expect(onHighlightItem.mock.calls[0][1]).toEqual(2);
    expect(snapshot(component)).toMatchSnapshot();
});

test('Down arrow key moves highlighted item down to next highlight-able item when list is active', () => {
    const onHighlightItem = jest.fn();
    component = mountList({
        active: true,
        children: getListItems().concat(getListItems()),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
        highlightedIndex: 2,
        onHighlightItem,
    });
    simulateDownArrowKey(component);
    expect(onHighlightItem).toHaveBeenCalledTimes(1);
    expect(onHighlightItem.mock.calls[0][1]).toEqual(4);
    expect(snapshot(component)).toMatchSnapshot();
});

test('Enter key selects the current highlighted item when list is active', () => {
    const onSelectItem = jest.fn();
    component = mountList({
        active: true,
        children: getListItems(),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
        highlightedIndex: 1,
        onSelectItem,
    });
    simulateEnterKey(component);
    expect(onSelectItem).toHaveBeenCalledTimes(1);
    expect(onSelectItem.mock.calls[0][1]).toEqual(1);
    expect(snapshot(component)).toMatchSnapshot();
});

test('List ignores keyUp events  when list is not active', () => {
    const onHighlightItem = jest.fn();
    component = mountList({
        children: getListItems(),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
        highlightedIndex: 2,
        onHighlightItem,
    });
    simulateUpArrowKey(component);
    expect(onHighlightItem).not.toHaveBeenCalled();
    simulateDownArrowKey(component);
    expect(onHighlightItem).not.toHaveBeenCalled();
    simulateEnterKey(component);
    expect(onHighlightItem).not.toHaveBeenCalled();
});

test('List selects item when clicking on a ListItem', () => {
    const onSelectItem = jest.fn();
    component = mountList({
        children: getListItems(),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
        highlightedIndex: 2,
        onSelectItem,
    });
    simulateClickItem(component, evt);
    expect(onSelectItem).toHaveBeenCalled();
    expect(onSelectItem.mock.calls[0][1]).toEqual(2);
});

test('List responds to mouseenter and mouseleave events', () => {
    const onMouseEnter = jest.fn();
    const onMouseLeave = jest.fn();
    component = renderList({
        children: getListItems(),
        highlightBackgroundColor: 'blue',
        highlightColor: 'white',
        highlightedIndex: 2,
        onMouseEnter,
        onMouseLeave
    });

    simulateMouseEnter(component, evt);
    expect(onMouseEnter).toHaveBeenCalledTimes(1);
    expect(onMouseEnter).toHaveBeenCalledWith(evt);

    simulateMouseLeave(component, evt);
    expect(onMouseEnter).toHaveBeenCalledTimes(1);
    expect(onMouseEnter).toHaveBeenCalledWith(evt);
});

function getListItems() {
    return [<SubHeader>Hello world</SubHeader>]
        .concat([
            'List',
            'Items'
        ]
            .map((child, index) => (
                <ListItem
                    itemOid={index}
                >
                    {child}
                </ListItem>
            )));
}
function simulateHover(wrapper, evt = {}) {
    wrapper.find('ListItem').at(1).simulate('mouseenter', evt);
}
function simulateMouseLeave(wrapper, evt = {}) {
    wrapper.find('ListItem').at(1).simulate('mouseleave', evt);
}
function simulateUpArrowKey(wrapper) {
    wrapper.find('ListItem').at(1).simulate('keyup', {keyCode: 38});
}
function simulateDownArrowKey(wrapper) {
    wrapper.find('ListItem').at(1).simulate('keyup', {keyCode: 40});
}
function simulateEnterKey(wrapper) {
    wrapper.find('ListItem').at(1).simulate('keyup', {keyCode: 13});
}
function simulateMouseEnter(wrapper, evt = {}) {
    wrapper.simulate('mouseenter', evt);
}
function simulateClickItem(wrapper, evt = {}) {
    wrapper.find('ListItem').at(1).simulate('click', evt);
}

