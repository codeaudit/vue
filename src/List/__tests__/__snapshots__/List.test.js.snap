exports[`test Down arrow key moves highlighted item down from the current highlighted item when list is active 1`] = `
<Connect(UI)
  active={true}
  highlightBackgroundColor="blue"
  highlightColor="white"
  highlightedIndex={1}
  onItemHighlighted={[Function]}>
  <UI
    active={true}
    highlightBackgroundColor="blue"
    highlightColor="white"
    highlightedIndex={1}
    massUpdateUI={[Function]}
    mountUI={[Function]}
    onItemHighlighted={[Function]}
    setDefaultUI={[Function]}
    ui={
      Object {
        "List": Object {
          "highlightedIndex": 2,
        },
        "__reducers": Object {},
      }
    }
    unmountUI={[Function]}
    updateUI={[Function]}>
    <List
      active={true}
      highlightBackgroundColor="blue"
      highlightColor="white"
      highlightedIndex={1}
      massUpdateUI={[Function]}
      mountUI={[Function]}
      onItemHighlighted={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onSelectItem={[Function]}
      resetUI={[Function]}
      setDefaultUI={[Function]}
      ui={
        Object {
          "highlightedIndex": 2,
        }
      }
      uiKey="List"
      uiPath={
        Array [
          "List",
        ]
      }
      unmountUI={[Function]}
      updateUI={[Function]}>
      <div
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "fontFamily": "\'Proxima Nova\', \'Lucida Sans Unicode\', \'Lucida Grande\', sans-serif",
            "fontSize": 14,
            "maxHeight": false,
            "overflow": "auto",
          }
        }>
        <div
          onClick={[Function]}>
          <SubHeader>
            <header
              data-radium={true}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "borderBottom": "1px dashed #a6a6a6",
                  "color": "#a6a6a6",
                  "padding": "6px 12px",
                  "textTransform": "uppercase",
                }
              }>
              Hello world
            </header>
          </SubHeader>
        </div>
        <div
          onClick={[Function]}>
          <ListItem
            highlightBackgroundColor="blue"
            highlightColor="white"
            highlighted={false}
            itemOid={0}
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            tabIndex={1}>
            <div
              data-radium={true}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "initial",
                  "cursor": "pointer",
                  "padding": "6px 12px",
                }
              }>
              <EventListener
                onKeyUp={[Function]}
                target="window" />
              List
            </div>
          </ListItem>
        </div>
        <div
          onClick={[Function]}>
          <ListItem
            highlightBackgroundColor="blue"
            highlightColor="white"
            highlighted={true}
            itemOid={1}
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            tabIndex={2}>
            <div
              data-radium={true}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "backgroundColor": "blue",
                  "color": "white",
                  "cursor": "pointer",
                  "padding": "6px 12px",
                }
              }>
              <EventListener
                onKeyUp={[Function]}
                target="window" />
              Items
            </div>
          </ListItem>
        </div>
      </div>
    </List>
  </UI>
</Connect(UI)>
`;

exports[`test Enter key selects the current highlighted item when list is active 1`] = `
<Connect(UI)
  active={true}
  highlightBackgroundColor="blue"
  highlightColor="white"
  highlightedIndex={1}
  onSelectItem={[Function]}>
  <UI
    active={true}
    highlightBackgroundColor="blue"
    highlightColor="white"
    highlightedIndex={1}
    massUpdateUI={[Function]}
    mountUI={[Function]}
    onSelectItem={[Function]}
    setDefaultUI={[Function]}
    ui={
      Object {
        "List": Object {
          "highlightedIndex": 2,
        },
        "__reducers": Object {},
      }
    }
    unmountUI={[Function]}
    updateUI={[Function]}>
    <List
      active={true}
      highlightBackgroundColor="blue"
      highlightColor="white"
      highlightedIndex={1}
      massUpdateUI={[Function]}
      mountUI={[Function]}
      onItemHighlighted={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onSelectItem={[Function]}
      resetUI={[Function]}
      setDefaultUI={[Function]}
      ui={
        Object {
          "highlightedIndex": 2,
        }
      }
      uiKey="List"
      uiPath={
        Array [
          "List",
        ]
      }
      unmountUI={[Function]}
      updateUI={[Function]}>
      <div
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "fontFamily": "\'Proxima Nova\', \'Lucida Sans Unicode\', \'Lucida Grande\', sans-serif",
            "fontSize": 14,
            "maxHeight": false,
            "overflow": "auto",
          }
        }>
        <div
          onClick={[Function]}>
          <SubHeader>
            <header
              data-radium={true}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "borderBottom": "1px dashed #a6a6a6",
                  "color": "#a6a6a6",
                  "padding": "6px 12px",
                  "textTransform": "uppercase",
                }
              }>
              Hello world
            </header>
          </SubHeader>
        </div>
        <div
          onClick={[Function]}>
          <ListItem
            highlightBackgroundColor="blue"
            highlightColor="white"
            highlighted={false}
            itemOid={0}
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            tabIndex={1}>
            <div
              data-radium={true}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "initial",
                  "cursor": "pointer",
                  "padding": "6px 12px",
                }
              }>
              <EventListener
                onKeyUp={[Function]}
                target="window" />
              List
            </div>
          </ListItem>
        </div>
        <div
          onClick={[Function]}>
          <ListItem
            highlightBackgroundColor="blue"
            highlightColor="white"
            highlighted={true}
            itemOid={1}
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            tabIndex={2}>
            <div
              data-radium={true}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "backgroundColor": "blue",
                  "color": "white",
                  "cursor": "pointer",
                  "padding": "6px 12px",
                }
              }>
              <EventListener
                onKeyUp={[Function]}
                target="window" />
              Items
            </div>
          </ListItem>
        </div>
      </div>
    </List>
  </UI>
</Connect(UI)>
`;

exports[`test List can have a specified highlight color for its ListItems 1`] = `
<UI
  highlightBackgroundColor="blue"
  highlightColor="white"
  massUpdateUI={[Function]}
  mountUI={[Function]}
  setDefaultUI={[Function]}
  ui={
    Object {
      "List": Object {
        "highlightedIndex": null,
      },
      "__reducers": Object {},
    }
  }
  unmountUI={[Function]}
  updateUI={[Function]}>
  <SubHeader>
    Hello world
  </SubHeader>
  <ListItem
    highlighted={false}
    itemOid={0}
    onClick={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}>
    List
  </ListItem>
  <ListItem
    highlighted={false}
    itemOid={1}
    onClick={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}>
    Items
  </ListItem>
</UI>
`;

exports[`test List can render ListItems 1`] = `
<UI
  massUpdateUI={[Function]}
  mountUI={[Function]}
  setDefaultUI={[Function]}
  ui={
    Object {
      "__reducers": Object {},
    }
  }
  unmountUI={[Function]}
  updateUI={[Function]}>
  <SubHeader>
    Hello world
  </SubHeader>
  <ListItem
    highlighted={false}
    itemOid={0}
    onClick={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}>
    List
  </ListItem>
  <ListItem
    highlighted={false}
    itemOid={1}
    onClick={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}>
    Items
  </ListItem>
</UI>
`;

exports[`test List can render ListItems with scroll bar 1`] = `
<Connect(UI)
  maxHeight={50}>
  <UI
    massUpdateUI={[Function]}
    maxHeight={50}
    mountUI={[Function]}
    setDefaultUI={[Function]}
    ui={
      Object {
        "List": Object {
          "highlightedIndex": null,
        },
        "__reducers": Object {},
      }
    }
    unmountUI={[Function]}
    updateUI={[Function]}>
    <List
      active={false}
      highlightBackgroundColor="#262626"
      highlightColor="#fff"
      massUpdateUI={[Function]}
      maxHeight={50}
      mountUI={[Function]}
      onItemHighlighted={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onSelectItem={[Function]}
      resetUI={[Function]}
      setDefaultUI={[Function]}
      ui={
        Object {
          "highlightedIndex": null,
        }
      }
      uiKey="List"
      uiPath={
        Array [
          "List",
        ]
      }
      unmountUI={[Function]}
      updateUI={[Function]}>
      <div
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "fontFamily": "\'Proxima Nova\', \'Lucida Sans Unicode\', \'Lucida Grande\', sans-serif",
            "fontSize": 14,
            "maxHeight": "50px",
            "overflow": "auto",
          }
        }>
        <div
          onClick={[Function]}>
          <SubHeader>
            <header
              data-radium={true}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "borderBottom": "1px dashed #a6a6a6",
                  "color": "#a6a6a6",
                  "padding": "6px 12px",
                  "textTransform": "uppercase",
                }
              }>
              Hello world
            </header>
          </SubHeader>
        </div>
        <div
          onClick={[Function]}>
          <ListItem
            highlightBackgroundColor="#262626"
            highlightColor="#fff"
            highlighted={false}
            itemOid={0}
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            tabIndex={1}>
            <div
              data-radium={true}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "initial",
                  "cursor": "pointer",
                  "padding": "6px 12px",
                }
              }>
              <EventListener
                onKeyUp={[Function]}
                target="window" />
              List
            </div>
          </ListItem>
        </div>
        <div
          onClick={[Function]}>
          <ListItem
            highlightBackgroundColor="#262626"
            highlightColor="#fff"
            highlighted={false}
            itemOid={1}
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            tabIndex={2}>
            <div
              data-radium={true}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "initial",
                  "cursor": "pointer",
                  "padding": "6px 12px",
                }
              }>
              <EventListener
                onKeyUp={[Function]}
                target="window" />
              Items
            </div>
          </ListItem>
        </div>
      </div>
    </List>
  </UI>
</Connect(UI)>
`;

exports[`test List's highlighted item can be set 1`] = `
<UI
  highlightBackgroundColor="blue"
  highlightColor="white"
  massUpdateUI={[Function]}
  mountUI={[Function]}
  setDefaultUI={[Function]}
  ui={
    Object {
      "List": Object {
        "highlightedIndex": null,
      },
      "__reducers": Object {},
    }
  }
  unmountUI={[Function]}
  updateUI={[Function]}>
  <SubHeader>
    Hello world
  </SubHeader>
  <ListItem
    highlighted={false}
    itemOid={0}
    onClick={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}>
    List
  </ListItem>
  <ListItem
    highlighted={false}
    itemOid={1}
    onClick={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}>
    Items
  </ListItem>
</UI>
`;

exports[`test List's list items are highlighted when hovered over 1`] = `
<UI
  highlightBackgroundColor="blue"
  highlightColor="white"
  massUpdateUI={[Function]}
  mountUI={[Function]}
  setDefaultUI={[Function]}
  ui={
    Object {
      "List": Object {
        "highlightedIndex": null,
      },
      "__reducers": Object {},
    }
  }
  unmountUI={[Function]}
  updateUI={[Function]}>
  <SubHeader>
    Hello world
  </SubHeader>
  <ListItem
    highlighted={false}
    itemOid={0}
    onClick={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}>
    List
  </ListItem>
  <ListItem
    highlighted={false}
    itemOid={1}
    onClick={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}>
    Items
  </ListItem>
</UI>
`;

exports[`test Up arrow key moves highlighted item up from the current highlighted item when list is active 1`] = `
<Connect(UI)
  active={true}
  highlightBackgroundColor="blue"
  highlightColor="white"
  highlightedIndex={2}
  onItemHighlighted={[Function]}>
  <UI
    active={true}
    highlightBackgroundColor="blue"
    highlightColor="white"
    highlightedIndex={2}
    massUpdateUI={[Function]}
    mountUI={[Function]}
    onItemHighlighted={[Function]}
    setDefaultUI={[Function]}
    ui={
      Object {
        "List": Object {
          "highlightedIndex": 1,
        },
        "__reducers": Object {},
      }
    }
    unmountUI={[Function]}
    updateUI={[Function]}>
    <List
      active={true}
      highlightBackgroundColor="blue"
      highlightColor="white"
      highlightedIndex={2}
      massUpdateUI={[Function]}
      mountUI={[Function]}
      onItemHighlighted={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onSelectItem={[Function]}
      resetUI={[Function]}
      setDefaultUI={[Function]}
      ui={
        Object {
          "highlightedIndex": 1,
        }
      }
      uiKey="List"
      uiPath={
        Array [
          "List",
        ]
      }
      unmountUI={[Function]}
      updateUI={[Function]}>
      <div
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "fontFamily": "\'Proxima Nova\', \'Lucida Sans Unicode\', \'Lucida Grande\', sans-serif",
            "fontSize": 14,
            "maxHeight": false,
            "overflow": "auto",
          }
        }>
        <div
          onClick={[Function]}>
          <SubHeader>
            <header
              data-radium={true}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "borderBottom": "1px dashed #a6a6a6",
                  "color": "#a6a6a6",
                  "padding": "6px 12px",
                  "textTransform": "uppercase",
                }
              }>
              Hello world
            </header>
          </SubHeader>
        </div>
        <div
          onClick={[Function]}>
          <ListItem
            highlightBackgroundColor="blue"
            highlightColor="white"
            highlighted={true}
            itemOid={0}
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            tabIndex={1}>
            <div
              data-radium={true}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "backgroundColor": "blue",
                  "color": "white",
                  "cursor": "pointer",
                  "padding": "6px 12px",
                }
              }>
              <EventListener
                onKeyUp={[Function]}
                target="window" />
              List
            </div>
          </ListItem>
        </div>
        <div
          onClick={[Function]}>
          <ListItem
            highlightBackgroundColor="blue"
            highlightColor="white"
            highlighted={false}
            itemOid={1}
            onClick={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            tabIndex={2}>
            <div
              data-radium={true}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "initial",
                  "cursor": "pointer",
                  "padding": "6px 12px",
                }
              }>
              <EventListener
                onKeyUp={[Function]}
                target="window" />
              Items
            </div>
          </ListItem>
        </div>
      </div>
    </List>
  </UI>
</Connect(UI)>
`;
